splitgraph/engine/postgres/engine.py

```diff
@@ -59,6 +59,8 @@
     # the connection property otherwise
     from psycopg2._psycopg import connection as Connection
 
+psycopg2.extensions.register_adapter(dict, Json)
```

Comment:
Note that this changes the behaviour globally for anything that will import this module -- the tests pass so it seems fine but it can lead to confusing bugs where whether or not you need to wrap your dict with `Json` will change depending on whether or not you imported this module (basically, not sure if this breaks other places that insert `Json(some_dict)`).